csharp/src/elements/Geometry/Polygon.cs

```diff
@@ -140,31 +325,124 @@ public bool Disjoint(Polygon polygon)
         }
 
         /// <summary>
-        /// Tests if this Polygon and the supplied Polygon share areas within their perimeters.
+        /// Tests if any of the supplied Polygons are coincident with this Polygon.
         /// </summary>
-        /// <param name="polygon">The Polygon to compare.</param>
+        /// <param name="polygons">The collection of Polygons to compare.</param>
         /// <returns>
-        /// Returns true if any vertex of either Polygon is within the perimeter of the other.
+        /// Returns true if any of the supplied Polygons do not intersect or touch.
         /// </returns>
-        public bool Intersects(Polygon polygon)
+        public bool Disjoint(IList<Polygon> polygons)
         {
             var thisPath = this.ToClipperPath();
-            var polyPath = polygon.ToClipperPath();
-            foreach (IntPoint vertex in thisPath)
+            foreach (Polygon polygon in polygons)
             {
-                if (Clipper.PointInPolygon(vertex, polyPath) == 1)
+                var polyPath = polygon.ToClipperPath();
+                foreach (IntPoint vertex in thisPath)
                 {
-                    return true;
+                    if (Clipper.PointInPolygon(vertex, polyPath) != 0)
+                    {
+                        return false;
+                    }
+                }
+                foreach (IntPoint vertex in polyPath)
+                {
+                    if (Clipper.PointInPolygon(vertex, thisPath) != 0)
+                    {
+                        return false;
+                    }
                 }
             }
-            foreach (IntPoint vertex in polyPath)
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if the supplied Polygon shares areas with this Polygon.
+        /// </summary>
+        /// <param name="polygon">The Polygon to compare with this Polygon.</param>
+        /// <returns>
+        /// Returns true if the two Polygons intersect at least once.
+        /// </returns>
+        public bool Intersects(Polygon polygon)
+        {
+            if (polygon == null)
+            {
+                return false;
+            }
+            var clipper = new Clipper();
+            var solution = new List<List<IntPoint>>();
+            clipper.AddPath(this.ToClipperPath(), PolyType.ptSubject, true);
+            clipper.AddPath(polygon.ToClipperPath(), PolyType.ptClip, true);
+            clipper.Execute(ClipType.ctIntersection, solution);
+            return solution.Count != 0;
+        }
+
+        /// <summary>
+        /// Tests if any of the supplied Polygons share areas with this Polygon.
+        /// </summary>
+        /// <param name="polygons">The list of Polygons to compare.</param>
+        /// <returns>
+        /// Returns true if any of the supplied Polygons intersects with this Polygon.
+        /// </returns>
+        public bool Intersects(IList<Polygon> polygons)
```

Comment:
As above, use singular `Interesects` in this method.

Reply:
Done.